concrete/src/Mail/MailServiceProvider.php

```diff
@@ -2,18 +2,25 @@
 namespace Concrete\Core\Mail;
 
 use Concrete\Core\Foundation\Service\Provider as ServiceProvider;
+use Concrete\Core\Mail\Transport\Smtp as SmtpTransport;
 
 class MailServiceProvider extends ServiceProvider
 {
     public function register()
     {
-        $register = array(
+        $app = $this->app;
+
+        $register = [
             'helper/mail' => '\Concrete\Core\Mail\Service',
             'mail' => '\Concrete\Core\Mail\Service',
-        );
+        ];
 
         foreach ($register as $key => $value) {
-            $this->app->bind($key, $value);
+            $app->bind($key, $value);
         }
+
+        $this->app->singleton(SmtpTransport::class, function () use ($app) {
```

Comment:
From what I can see, there's no reason to wrap the zend smtp transport. Just mark it as a singleton and use 
```php
$app->when(Smtp::class)->needs(SmtpOptions::class)->give(function($app) {
    // build your options object here using $app->make
});
```